crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -166,14 +168,73 @@ impl Plugin for MeshRenderPlugin {
     }
 }
 
-#[derive(Component, ShaderType, Clone)]
+#[derive(Component)]
+pub struct MeshTransforms {
+    pub transform: Affine3A,
```

Comment:
If we're now doing nontrivial transformation into MeshUniform already, can we try using Affine3 over Affine3A here? We could save more CPU side memory and reduce command time costs.

Reply:
There is no `glam::Affine3`. It looks like we only use the translation from the transform for view z calculations in the render app. I could then change them out for our own implementation of Affine3 that just has conversion to Affine3A implemented in case someone needs to do calculations with it.

Reply:
I did that and retested (M1 Max, on AC power and fully charged, `many_cubes -- sphere` for 1500 frames).

main (yellow) vs PR (red):
frame time:
![Screenshot 2023-08-12 at 12 52 55](https://github.com/bevyengine/bevy/assets/302146/1c1d614e-1a87-40ad-a5b1-08c157d0edf0)
extract_meshes system:
![Screenshot 2023-08-12 at 12 53 35](https://github.com/bevyengine/bevy/assets/302146/eb5dbdb2-2986-42f0-9bcb-4078723d0b66)
extract_meshes system commands:
![Screenshot 2023-08-12 at 12 53 43](https://github.com/bevyengine/bevy/assets/302146/52555759-8dd5-41ac-aa19-1e5f56446302)

main (yellow) vs PR with `MeshTransforms` using custom `Affine3`:
frame time:
![Screenshot 2023-08-12 at 12 53 57](https://github.com/bevyengine/bevy/assets/302146/837b9e6c-e53d-427f-a6cf-0861aae22709)
extract_meshes system:
![Screenshot 2023-08-12 at 12 54 11](https://github.com/bevyengine/bevy/assets/302146/0a3bbcb5-a9ed-42c6-a37e-c74f0d7b555a)
extract_meshes system commands:
![Screenshot 2023-08-12 at 12 54 26](https://github.com/bevyengine/bevy/assets/302146/0ac6189a-93db-4f20-92b4-e60eebabdf2c)

Summary:
- frame times vs main in median frame time for this test run:
  - PR gives 6% reduction
  - PR + Affine3 gives 14% reduction

So it does indeed seem like adding `struct Affine3 { matrix3: Mat3, translation: Vec3 }` is worth it. Done.